upgrade/upgrade-scripts/rootfs-cleanup

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import subprocess
+from typing import Any, List, Type
+
+IMAGE_PATH = os.path.dirname(os.path.realpath(__file__))
+
+
+#
+# Helper to compare two version strings using "dpkg --compare-versions".
+#
+def dpkgcmp(vers1: str, op: str, vers2: str) -> bool:
+    return subprocess.run(["dpkg", "--compare-versions", vers1, op,
+                           vers2]).returncode == 0
+
+
+#
+# Helper to obtain the "current-version" of a rootfs filesytem/snapshot.
+#
+def version(rootfs: str) -> str:
+    return str(
+        subprocess.check_output([
+            "zfs", "get", "-Hpo", "value", "com.delphix:current-version", rootfs
+        ]).decode("UTF-8").strip())
+
+
+#
+# Helper class used when sorting rootfs filesystems and snapshots. This allows
+# us to use the "sorted" builtin, to build a list of all rootfs filesystems and
+# snapshots on the system, sorted baseed on the "current-version" property.
+#
+class rootfscmp(object):
+
+    def __init__(self, rootfs: str, *args: List[Any]):
+        self.rootfs = rootfs
+
+    def __lt__(self, other: Type["rootfscmp"]) -> bool:
+        return dpkgcmp(version(self.rootfs), "lt", version(other.rootfs))
+
+    def __gt__(self, other: Type["rootfscmp"]) -> bool:
+        return dpkgcmp(version(self.rootfs), "gt", version(other.rootfs))
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, rootfscmp):
+            return NotImplemented
+        return dpkgcmp(version(self.rootfs), "eq", version(other.rootfs))
+
+    def __le__(self, other: Type["rootfscmp"]) -> bool:
+        return dpkgcmp(version(self.rootfs), "le", version(other.rootfs))
+
+    def __ge__(self, other: Type["rootfscmp"]) -> bool:
+        return dpkgcmp(version(self.rootfs), "ge", version(other.rootfs))
+
+    def __ne__(self, other: object) -> bool:
+        if not isinstance(other, rootfscmp):
+            return NotImplemented
+        return dpkgcmp(version(self.rootfs), "ne", version(other.rootfs))
+
+
+def main() -> None:
+    snapshots = subprocess.check_output([
+        "zfs", "list", "-r", "rpool/ROOT", "-d", "2", "-Ho", "name", "-t",
+        "snapshot"
+    ]).decode("UTF-8").splitlines()
+
+    for snap in snapshots:
+        name = snap.split("@")[1]
+
+        #
+        # We filter out any container-delphix snapshots, as those removed by
+        # the calls to "rootfs-container delete", on the relevant filesystems.
+        #
+        if re.search("^container-delphix.\w{7}$", name):
+            snapshots.remove(snap)
+            continue
+
+        #
+        # Out of an abundance of caution, verify the snapshot name matches
+        # our expectations.
+        #
+        if not re.search("^execute-upgrade.\w{7}$", name):
```

Comment:
both flake8 and pylint seem to be complaining about `\w`:
flake8:
```
upgrade/upgrade-scripts/rootfs-cleanup:83:44: W605 invalid escape sequence '\w'
```
pylint:
```
upgrade/upgrade-scripts/rootfs-cleanup:83:17: W1401: Anomalous backslash in string: '\w'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
```

Not sure but I think you may need to use `\\w` instead

Reply:
I've tested this to work as-is.

Reply:
Yeah looks like this syntax still works but is deprecated and will generate a syntax error in the future: see https://www.flake8rules.com/rules/W605.html

Reply:
Additionally, when I run `flake8` locally, I don't get that warning...

Reply:
OK, I've added `r` to those two strings.

Reply:
Yeah it seems to be introduced in python 3.6, perhaps your `flake8` is linked against python 2?

Reply:
not sure:
```
± python3 --version
Python 3.6.9

± python3 -m flake8 --version
3.9.1 (mccabe: 0.6.1, pycodestyle: 2.7.0, pyflakes: 2.3.1) CPython 3.6.9 on
Linux
```
I don't think it's important, though.. I went ahead and added `r`, which I think is correct, and shouldn't really change much w.r.t. testing or anything.